lib/proxy/helpers.rb

```diff
@@ -7,16 +7,18 @@ module Proxy::Helpers
   # Accepts a html error code and a message, which is then returned to the caller after adding to the proxy log
   # OR  a block which is executed and its errors handled in a similar way.
   # If no code is supplied when the block is declared then the html error used is 400.
-  def log_halt code=nil, exception=nil
-    message = exception.is_a?(String) ? exception : exception.to_s
+  def log_halt code=nil, exception_or_msg=nil, custom_msg=nil
```

Comment:
I never knew the second argument is an exception. I *always* thought it was a message since that's pretty much used everywhere. To me a signature this signature feels more natural but not sure if this is compatible:
```suggestion
  def log_halt(code=nil, message=nil, exception=nil)
```
I also find it much easier to read with braces, but that may be a personal preference.

Lastly, should code default to 400 instead of inside the body?

Reply:
Yeah it's a top secret. I wish I could change this, but almost every plugin use this helper...